### DISCUSSION TITLE
Alter repository

### DISCUSSION THREAD

**Discussion by TFS-HOS** [apache/doris#40889] [A - General / Q&A]
Good day, 
My s3 keys need to be renewed every 7 days, how can I update a repository to use the new keys?

Tried this:
```
ALTER REPOSITORY my_repo
SET (
    "s3.access_key" = "mykey",
    "s3.secret_key" = "mysecret"
);
```

**Comment by ixzc** [comment]
it's not support now. we plan to support this in the future.

**Reply by morningman** [reply]
But you can try drop and re-create repository every 7 days.
drop repository will not drop backup data on s3.

**Reply by TFS-HOS** [reply]
> But you can try drop and re-create repository every 7 days. drop repository will not drop backup data on s3.

That works in the meantime, but for simply changing the keys it is excessive. Looking forward to the future support for this.

**Reply by ygsa** [reply]
Any updates?
